scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/link-list-handler.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Check text nodes at the top level of the static_methods section.
+ * Text nodes here may only contain newlines.
+ */
+function checkTextNode(node, logError) {
+  const newlinesOnly = /^\n*$/;
+  if (!node.value.match(newlinesOnly)) {
+    logError(
+      "Text nodes in section may only contain newlines",
+      "text-nodes-in-section"
+    );
+  }
+}
+
+/**
+ * Check elements at the top level of the static_methods section.
+ * Only `dl` elements are allowed.
+ */
+function checkElementNode(node, logError) {
+  if (node.tagName !== "dl") {
+    logError(
+      "Section may only contain DL elements",
+      "non-dl-elements-in-section"
+    );
+    return;
+  }
+  checkDL(node, logError);
+}
+
+/**
+ * Check the structure of a `dl` element that represents a list of links.
+ */
+function checkDL(node, logError) {
+  let title = "";
+  for (const child of node.children) {
+    // We are only going to check `dt` elements, and will tolerate
+    // any others
```

Comment:
There are some utilities that might make traversing these trees a bit easier:

Using [`unist-util-visit`](https://github.com/syntax-tree/unist-util-visit#readme) or `hast-util-select`'s `selectAll` and `visit` might be a more idiomatic way to walk all these elements. Something like…

```javascript
visit(tree, node => node.tagName === "dt", node => /* … do something with each dt … */);
```

or

```javascript
const dts = selectAll('dt', tree);
for (const dt of dts) {
  visit(dt.children, node => {
    /* .. */
  });
}
```

[`unist-util-visit-parents`](https://github.com/syntax-tree/unist-util-visit-parents) will let you visit the innermost element you're looking for and then check bottom-up for the structure you're looking for:

```javascript
visit(dtNode, node => node.type === "text", (node, ancestors) => {
  /* assert !children.length and the immediate ancestors are `<code>` and `<a>` */
});
```

(There might be a util to get the leaf node(s) of a tree too, don't know off the top of my head.)